internal/envtest/envtest.go

```diff
@@ -101,13 +118,45 @@ func (e *Environment) stop() {
 
 func (e *Environment) CreateNamespace(t *testing.T) string {
 	namespace := strings.ToLower(t.Name())
-	if err := kubernetes.CreateNamespaceIfNotExists(e.intClient, namespace); err != nil {
+	if err := kubernetes.CreateNamespaceIfNotExists(e.Client, namespace); err != nil {
 		t.FailNow()
 	}
 
 	return namespace
 }
 
-func (e *Environment) RestConfig() *rest.Config {
-	return e.env.Config
+func (e *Environment) GetKubeconfig(t *testing.T) ([]byte, error) {
```

Comment:
question: Same naming convention, this would just as well as Kubeconfig().

Reply:
It would, but the problem is that this is actually creating a kubeconfig from the rest file. This is all dealt with under my other testing PR. Can we defer further changes until there so that we can refactor this properly if required. 

Otherwise that is going to end up in rebase hell, and we need to carefully consider when the Kubeconfig field actually gets populated, when operating under different backends.

Reply:
I don't feel like any PR should be merged at any point with reasons "we might have a next PR that actually fixes things". And this as a generic remark. 

Each PR should be standalone to provide something, not something that's buggy/broken and maybe fixed in the next one where one could then say "but this is already in the codebase, I'm not changing it here" thus making things super difficult to reason.

Do you disagree with this approach of reviews?